### DISCUSSION TITLE
Most Effective/Accepted Way to Measure Resource Consumption by K3s and Pods

### DISCUSSION THREAD

**Discussion by oheckmann** [k3s-io/k3s#6367] [Q&A]
Good Afternoon, I am trying to get some information about how K3s performs in certain situations, and part of this involves looking at how a k3s-system (both server and pods) are consuming resources in some certain situations.
I've tried Perf, but that (to my knowledge) only shows use of services in terms of overhead - which makes it difficult to compare performance in different situations. Is there a method or system that is generally accepted as the "best" method to find this information over a period of time? If not, are their methods/systems that you would recommend to use?

**Comment by brandond** [comment]
Scraping metrics with Prometheus and visualizing the data with Grafana is the generally-accepted way to monitor a Kubernetes cluster and the applications deployed to it.

**Reply by oheckmann** [reply]
Thank you
Are there any resources that you recommend to get started?
